File: LeetCode/2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        unsigned long sum = 0;
        int i = 0;
        for(int ch: chalk){
            sum += ch;
        }

        k %= sum;
        while(chalk[i] <= k){
            k -= chalk[i];
            i ++;
        }
        return i;
    }
};

File: LeetCode/2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        sum_of_chalks = sum(chalk)
        passes = k // sum_of_chalks
        remainder = k - passes*sum_of_chalks
        index = 0

        while remainder >= chalk[index]:
            remainder -= chalk[index]
            index += 1
        return index
